modules/core/arrow-typeclasses/src/main/kotlin/arrow/typeclasses/suspended/prelude.kt

```diff
@@ -0,0 +1,38 @@
+package arrow.typeclasses.suspended
+
+import arrow.Kind
+
+interface Prelude {
+  suspend fun <A> effectIdentity(a: A): A = a
+}
+
+interface BindSyntax<F> : Prelude {
+
+  suspend fun <A> Kind<F, A>.bind(): A
+
+  suspend operator fun <A> Kind<F, A>.component1(): A =
+    bind()
+
+  fun <A> f(fa: suspend () -> A): Kind<F, A>
+
+  fun <A> (suspend () -> A).k(): Kind<F, A> = f(this)
+
+  fun <A, B> (suspend (A) -> B).k(): (Kind<F, A>) -> Kind<F, B> =
+    { suspend { this(it.bind()) }.k() }
+
+  fun <A, B, C> (suspend (A, B) -> C).k(): (Kind<F, A>, Kind<F, B>) -> Kind<F, C> =
+    { ka, kb -> suspend { this(ka.bind(), kb.bind()) }.k() }
+
+  fun <A, B, C, D> (suspend (A, B, C) -> D).k(): (Kind<F, A>, Kind<F, B>, Kind<F, C>) -> Kind<F, D> =
+    { ka, kb, kc -> suspend { this(ka.bind(), kb.bind(), kc.bind()) }.k() }
+
+  fun <A, B> (suspend (A) -> B).kr(unit: Unit = Unit): (A) -> Kind<F, B> =
```

Comment:
what does `kr` mean here?

Reply:
`k` is `liftM` which we don't have in `Arrow` in monad, there is also `liftA` versions of it for applicative. https://www.haskell.org/hoogle/?hoogle=liftM . It supports N arities we can provide up to 3 or 4 because otherwise the number of methods in the extensions will increase considerably. I renamed them in this commit to `liftM` and `flatLiftM` until we come up with better names in https://github.com/arrow-kt/arrow/pull/1263/commits/cb6a3aa2af3beba7c96b7efea7c9d9458625bbc2

Reply:
Let's discuss over slack names for all the functions of the DSL